pruner/example_programs/fabric_with_creduce/fabric.p4

```diff
@@ -0,0 +1,8 @@
+#include "include/size.p4"
+#include "include/control/filtering.p4"
+#include "include/checksum.p4"
+#include "include/parser.p4"
+ bit<16> a (inout ipv4_t b){                          return b.total_len;                      }
+                         control c (inout parsed_headers_t d,                        inout fabric_metadata_t e,                        inout standard_metadata_t f                        ) {                         table g {                           key = {                           a(d.inner_ipv4): exact @name("");                        }                           actions = {     }                       }                          apply {                              g.apply();                       }                      }
+                         control h (inout parsed_headers_t d,                       inout fabric_metadata_t e,                       inout standard_metadata_t f) {             apply {      }                      }
+                         V1Switch(     FabricParser(),     FabricVerifyChecksum(),     c(),     h(),     FabricComputeChecksum(),     FabricDeparser() ) main;
```

Comment:
Is this how creduce reduces files? Looks terrible haha. Does this still  produce a bug?

Reply:
yep it looks terrible haha. Yes, it does still produce a bug.

Reply:
That is really efficient if it can actually do that.

Reply:
I agree, I was pretty surprised too.

Reply:
So what is the total reduction percentage? Might be hard to beat if the program manages to do this for all types of bugs.

Reply:
I think they use a slightly different metric than us. They report around 78% reduction rate. It might be based on the number of chars.

Reply:
But also, our pruner unrolls the includes in the program.

Reply:
Also, I think there is a huge difference in time of completion. Even though creduce is multi threaded, we are much much faster than them. I think that amounts to the more targeted approach, many of there passes seem to fail.